Fill in the profile form with values

vd1992 opened the pull request:
resolves #1996 

Some issues, per wireframe, there isn't anything in the schema yet for "job duration" or "disliked operational requirements"
Intl-ing and localizing ignored until the form is appropriately filled out
GraphQL stuff works fine I think locally

vd1992 commented:
From local environment: 

![image](https://user-images.githubusercontent.com/40485260/161147351-88bd3803-69a3-488a-b80a-853611c9310f.png)

vd1992 commented:
All needed fields pulled in now I think from Schema.
Functionality all present.

vd1992 commented:
You can use the Controls tool for Chromatic Storybook to pass in specific values if you have something in mind to test

petertgiles reviewed (changes requested):
Whoah, this is looking cool!  :tada: A few suggestions:

There are some issues with the page when no information is there (for example, when not logged in):
- Many boxes are missing the "There are required fields missing" message.
- Field labels with no entered data should not be displayed.
It would be good to have a storybook story in this state to be able to review.

The naming of the "inner component" and "outer component" is a little out of convention, I think.  I suggest _ProfilePage_ for the outer component and _ProfileForm_ for the inner component to be more like _SearchPage_ and _SearchForm_.  I think keeping them in the same file like this with this name is fine though.

Do we need the giant _ProfilePageProps_ interface?  This is the _User_ type, isn't it?

Regarding _skills and experience_, James' PR should be in soon.  I think it's worth holding this one back until that's in so that you can complete that section.

petertgiles reviewed (changes requested):
Please check all the enums, as well.  There are a few that are displaying as the raw enum instead of the localized description.

vd1992 commented:
> Whoah, this is looking cool! 🎉 A few suggestions:
> 
> There are some issues with the page when no information is there (for example, when not logged in):
> 
>     * Many boxes are missing the "There are required fields missing" message.
> 
>     * Field labels with no entered data should not be displayed.
>       It would be good to have a storybook story in this state to be able to review.
> 
> 
> The naming of the "inner component" and "outer component" is a little out of convention, I think. I suggest _ProfilePage_ for the outer component and _ProfileForm_ for the inner component to be more like _SearchPage_ and _SearchForm_. I think keeping them in the same file like this with this name is fine though.
> 
> Do we need the giant _ProfilePageProps_ interface? This is the _User_ type, isn't it?
> 
> Regarding _skills and experience_, James' PR should be in soon. I think it's worth holding this one back until that's in so that you can complete that section.

- Skill and Experiences is out of scope of the issue though
- Required boxes won't apply to Inclusion, and Government Info is missing that one field so that one will be pending. The other ones seem to have it appear. Not sure if they fail to appear in some cases though. 
- I don't know if I need the interface, so far I just try to avoid the red underlines through whatever means. 

Renamed components, empty fields hidden now, and added in Enum stuff. Standard Intl I'm leaving until I've made sure the page is complete with all the stuff I need to add.

vd1992 commented:
types tamed, type monstrosity eliminated

petertgiles commented on frontend/talentsearch/src/js/components/profile/ProfilePage/ProfilePage.tsx:
```diff
@@ -160,7 +269,25 @@ export const ProfilePage: React.FC<ProfilePageProps> = ({
+                data-h2-padding="b(all, m)"
+                data-h2-radius="b(s)"
+              >
+                {(candidateArray === null || candidateArray.length === 0) && (
```
You're doing a lot of `=== null` and `=== undefined` and `=== 0` in this file. These are all considered falsy in javascript.  You could using the more concise form of direct boolean evalution like `!candidateArray || !candidateArray.length` to shorten the code.  I see you doing this in other places on this file so I suggest that form to be consisent.

petertgiles commented on frontend/talentsearch/src/js/components/profile/ProfilePage/ProfilePage.tsx:
```diff
@@ -249,7 +648,64 @@ export const ProfilePage: React.FC<ProfilePageProps> = ({
+              >
+                {/* !!!!!!!!!!!!
+                TODO
+                ADD GOVERNMENT EMPLOYEE TYPE TO THIS FORM FUNCTION, THEN ADD EMPLOYEE TYPE LOGIC HERE
```
This is the missing field you mentioned?  Could you add the issue # here?

vd1992 replied:
going to mention this component in #2329

petertgiles commented on frontend/talentsearch/src/js/components/profile/ProfilePage/ProfilePage.tsx:
```diff
@@ -340,7 +920,74 @@ export const ProfilePage: React.FC<ProfilePageProps> = ({
+                      {isWoman && (
+                        <li data-h2-font-weight="b(700)">
+                          {intl.formatMessage({
+                            defaultMessage: "Woman",
```
I feel like we use these EE groups everywhere even though they're not a graphql enum.  I suggest adding them to the localized constants file.

petertgiles reviewed (changes requested):
Looking great!  I noticed a few more things: 

Checking null state:
- For _Diversity, equity and inclusion_ there should be a link for _employment equity groups_.
- Many of the boxes should have a _You haven't added any information here yet_ message.

For non null state:
- None of the stories have data for pool candidates.  Bug in component or query or seeder?
- For _Government information_ the items should be in an unordered list (dots)
- For the _About me_ section could you make the two columns collapse for small screens?
![image](https://user-images.githubusercontent.com/8978655/161837224-82caccfb-62ff-4640-8d32-be7c17bc9b99.png)

vd1992 commented:
will edit comment this later

petertgiles reviewed (approved):
Nice work, Vachan.  It looks great!

vd1992 commented:
> Annoyingly pedantic comment: Should the component be called ProfileForm? It doesn't contain a form! ;)

Peter suggested it above, so I rolled with it!
"The naming of the "inner component" and "outer component" is a little out of convention, I think. I suggest ProfilePage for the outer component and ProfileForm for the inner component to be more like SearchPage and SearchForm. I think keeping them in the same file like this with this name is fine though."
